src/components/DashboardPage/DashboardPage.tsx

```diff
@@ -0,0 +1,270 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, { MouseEventHandler, useCallback, useEffect, useState } from 'react';
+import dynamic from 'next/dynamic';
+import { useRouter } from 'next/router';
+import { nullable, Button } from '@taskany/bricks';
+
+import { refreshInterval } from '../../utils/config';
+import { ExternalPageProps } from '../../utils/declareSsrProps';
+import { ModalEvent, dispatchModalEvent } from '../../utils/dispatchModal';
+import { createFilterKeys } from '../../utils/hotkeys';
+import { useUrlFilterParams } from '../../hooks/useUrlFilterParams';
+import { useFilterResource } from '../../hooks/useFilterResource';
+import { routes } from '../../hooks/router';
+import { Page, PageContent } from '../Page';
+import { CommonHeader } from '../CommonHeader';
+import { FiltersPanel } from '../FiltersPanel/FiltersPanel';
+import { GoalsGroup } from '../GoalsGroup';
+import { GoalsListContainer } from '../GoalListItem';
+import { PageTitle } from '../PageTitle';
+import { Nullish } from '../../types/void';
+import { trpc } from '../../utils/trpcClient';
+import { FilterById, GoalByIdReturnType, ProjectByIdReturnType } from '../../../trpc/inferredTypes';
+import { ProjectItemStandalone } from '../ProjectListItem';
+
+import { tr } from './DashboardPage.i18n';
+
+const GoalPreview = dynamic(() => import('../GoalPreview/GoalPreview'));
+const ModalOnEvent = dynamic(() => import('../ModalOnEvent'));
+const FilterCreateForm = dynamic(() => import('../FilterCreateForm/FilterCreateForm'));
+const FilterDeleteForm = dynamic(() => import('../FilterDeleteForm/FilterDeleteForm'));
+
+export const DashboardPage = ({ user, ssrTime, locale }: ExternalPageProps) => {
+    const router = useRouter();
+    const [preview, setPreview] = useState<GoalByIdReturnType | null>(null);
+    const { toggleFilterStar } = useFilterResource();
+
+    const utils = trpc.useContext();
+
+    const presetData = trpc.filter.getById.useQuery(router.query.filter as string, { enabled: !!router.query.filter });
+
+    const {
+        currentPreset,
+        queryState,
+        queryString,
+        setPriorityFilter,
+        setStateFilter,
+        setTagsFilter,
+        setTagsFilterOutside,
+        setEstimateFilter,
+        setIssuerFilter,
+        setOwnerFilter,
+        setParticipantFilter,
+        setProjectFilter,
+        setStarredFilter,
+        setWatchingFilter,
+        setSortFilter,
+        setFulltextFilter,
+        resetQueryState,
+        setPreset,
+    } = useUrlFilterParams({
+        preset: presetData?.data,
+    });
+
+    const { data, isLoading } = trpc.goal.getUserGoals.useQuery(queryState, {
+        keepPreviousData: true,
+        staleTime: refreshInterval,
+    });
+
+    const goals = data?.goals;
+    const meta = data?.meta;
```

Comment:
Don't we have "prefer destructurization" rule enabled?

Reply:
We have, but data may be null

Reply:
`const { goals, meta } = data || {}` ?

Reply:
Yep, why not. But this is not new code. It was moved from GoalsPage... Let's do it in https://github.com/taskany-inc/issues/issues/1096